src/lib.rs

```diff
@@ -93,7 +93,6 @@
 //! differently. But as a rule: if you know the size of the body, it's usually more efficient to
 //! declare it up front. But if you don't, things will still work.
 
-#![forbid(rust_2018_idioms)]
```

Comment:
We also deny warnings so that's the same as having it be an error. We also can't actually fix the code since Serde's codegen would need to be adjusted. I don't think there's a way out of this one unfortunately.